Spigot-Server-Patches/0594-Damage-with-any-cause.patch

```diff
@@ -0,0 +1,454 @@
++                        return DamageSource.witherSkull((EntityWitherSkull) attackingEntity, projectile);
++                    }
++                    return DamageSource.projectile(projectile, projectile.getShooter());
++                }
++                return DamageSource.GENERIC;
++            case SUFFOCATION:
++                return DamageSource.STUCK;
++            case FALL:
++                return DamageSource.FALL;
++            case FIRE:
++                return DamageSource.FIRE;
++            case FIRE_TICK:
++                return DamageSource.BURN;
++            case MELTING:
++                return CraftEventFactory.MELTING;
++            case LAVA:
++                return DamageSource.LAVA;
++            case DROWNING:
++                return DamageSource.DROWN;
++            case VOID:
++                return DamageSource.OUT_OF_WORLD;
++            case LIGHTNING:
++                return DamageSource.LIGHTNING;
++            case STARVATION:
++                return DamageSource.STARVE;
++            case POISON:
++                return CraftEventFactory.POISON;
++            case MAGIC:
++                return DamageSource.MAGIC;
++            case WITHER:
++                return DamageSource.WITHER;
++            case FALLING_BLOCK:
++                return DamageSource.FALLING_BLOCK;
++            case THORNS:
++                if (attackingEntity == null) {
++                    return DamageSource.GENERIC;
++                }
++                return DamageSource.thorns(attackingEntity);
++            case DRAGON_BREATH:
++                return DamageSource.DRAGON_BREATH;
++            case CUSTOM:
++                return DamageSource.GENERIC;
++            case FLY_INTO_WALL:
++                return DamageSource.FLY_INTO_WALL;
++            case HOT_FLOOR:
++                return DamageSource.HOT_FLOOR;
++            case CRAMMING:
++                return DamageSource.CRAMMING;
++            case DRYOUT:
++                return DamageSource.DRYOUT;
++        }
++        return DamageSource.GENERIC;
++    }
++
++}
+diff --git a/src/main/java/net/minecraft/server/DamageSource.java b/src/main/java/net/minecraft/server/DamageSource.java
+index 6fe5678cffc2487fe00c953d772f764bb37a4b11..6226935318368907a1f6d37e3f8c04e3be724f7e 100644
+--- a/src/main/java/net/minecraft/server/DamageSource.java
++++ b/src/main/java/net/minecraft/server/DamageSource.java
+@@ -1,5 +1,9 @@
+ package net.minecraft.server;
+ 
++import org.bukkit.craftbukkit.event.CraftEventFactory;
++import org.bukkit.entity.LightningStrike;
++import org.bukkit.event.entity.EntityDamageEvent;
++
+ import javax.annotation.Nullable;
+ 
+ public class DamageSource {
+@@ -150,6 +154,14 @@ public class DamageSource {
+         this.translationIndex = s;
+     }
+ 
++    public static DamageSource thorns(Entity attackingEntity) { return DamageSource.a(attackingEntity); } // Paper OBFHELPER
```

Comment:
obfhelpers should be directly above the methods they're pointing to

Reply:
I was told to keep diff minimal

Reply:
not at the cost of OBFHELPERS, idea is to minimise unneeded changes which just create noise, not stuff which is intended to help during patching, etc